src/dune_engine/sandbox.ml

```diff
@@ -71,6 +73,15 @@ let copy_recursively =
   in
   let rec loop item ~src ~dst =
     match (item : Item.t) with
+    | Link ->
+      (match Path.Untracked.stat_exn src with
+       | { Unix.st_kind = S_REG; _ } -> Io.copy_file ~chmod:chmod_file ~src ~dst ()
+       | { Unix.st_kind = S_DIR; st_perm = perms; _ } ->
+         loop (Directory { perms }) ~src ~dst
+       | { Unix.st_kind; _ } -> loop (Other st_kind) ~src ~dst
+       | exception Unix.Unix_error (ENOENT, _, _) ->
+         let contents = Unix.readlink (Path.to_string src) in
```

Comment:
What's this case doing? Trying to recreate a broken link in the sandbox? Shouldn't this be an error instead?

Reply:
Packages may create their own broken links for their own tests for example. I think its better to ignore this rather than let dune get stuck on it.

Reply:
Yes, that ends up being an issue in practice. Especially for packages that require broken symlinks to run their test suites. We could fork the few packages that do this, but package builds are complete black boxes to us so I think it will be easier to just support it.

Reply:
I should also note that we don't even support broken symlinks in data_only_dirs which is something I looked into before. It's not clear how to handle that case in fs_memo.

Reply:
Can we force the users to request this behaviour rather than it being the default? Such automagic, I think, always comes back to bite later.

Reply:
In other words, I think seeing a failure and reacting to it by picking a special sandboxing mode wastes a lot less time compared to the build silently succeeding with unexpected results.

Reply:
Umm, how about I'll split this behavior from the current PR if it's controversial.

> Can we force the users to request this behaviour rather than it being the default?

Sure, let's think about how. Would it be a sandboxing option? Seems like it's a bit more general and can apply to non sandboxed rules that produce directory targets. E.g. "allow_broken_symlinks_in_target_directories".